### ISSUE TITLE
Team Builder should filter pokemon by cup type

### ISSUE THREAD

**Issue by davidpeden3** [pvpoke/pvpoke#15]

**Comment by pvpoke** [comment]
When changing cups, the Pokemon selection's `<option>` tags are hidden via CSS. However this doesn't work on all devices. The search bar also doesn't take these hidden tags into account, so that at's an easy fix to start with. I'll get that updated.

What should probably happen is the dropdown should reconstruct its `<option>` tags from the gamemaster data whenever the cup is changed so the filter isn't reliant on CSS. The only tricky thing is you probably want to preserve the current selection whether it's in the selected cup or not.

**Comment by davidpeden3** [comment]
I agree the `<option>`s should be reconstructed (client-side via filtering) though I don't think you should maintain an actively selected pokemon if it's not valid for a given cup. From a usability standpoint, you could simply store the active selections per cup in local storage so that they are restored when they are no longer filtered.
